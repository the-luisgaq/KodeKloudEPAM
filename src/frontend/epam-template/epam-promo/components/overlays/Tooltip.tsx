import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

interface TooltipMods {
    /**
     * Defines component color.
     * @default 'gray'
     */
    color?: 'white' | 'gray' | 'red' | uui.TooltipProps['color'];
}

/** Represents the properties of a Tooltip component. */
export interface TooltipProps extends uui.TooltipCoreProps, TooltipMods {}

export const Tooltip = createSkinComponent<uui.TooltipProps, TooltipProps>(uui.Tooltip);
