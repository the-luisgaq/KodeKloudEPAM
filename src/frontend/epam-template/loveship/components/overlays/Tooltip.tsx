import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

interface TooltipMods {
    /**
     * Defines component color.
     * @default 'gray'
     */
    color?: 'white' | 'fire' | 'gray' | uui.TooltipProps['color'];
}

/** Represents the properties of the Tooltip component. */
export interface TooltipProps extends uui.TooltipCoreProps, TooltipMods {}

export const Tooltip = createSkinComponent<uui.TooltipProps, TooltipProps>(uui.Tooltip);
