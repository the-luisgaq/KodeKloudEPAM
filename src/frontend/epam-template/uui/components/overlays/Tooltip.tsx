import { Overwrite, withMods } from '@epam/uui-core';
import * as uuiComponents from '@epam/uui-components';
import css from './Tooltip.module.scss';

interface TooltipMods {
    /**
     * Tooltip color
     * @default 'inverted'
     */
    color?: 'neutral' | 'inverted' | 'critical';
}

export interface TooltipCoreProps extends uuiComponents.TooltipProps {}

export interface TooltipModsOverride {}

export interface TooltipProps extends TooltipCoreProps, Overwrite<TooltipMods, TooltipModsOverride> {}

function applyTooltipMods(mods: TooltipMods) {
    return [
        css.root,
        `uui-color-${mods.color || 'inverted'}`,
    ];
}

export const Tooltip = withMods<TooltipProps, TooltipProps>(uuiComponents.Tooltip, applyTooltipMods);
