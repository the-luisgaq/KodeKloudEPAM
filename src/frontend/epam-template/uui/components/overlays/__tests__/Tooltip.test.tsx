import React from 'react';
import { Tooltip } from '../Tooltip';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('Tooltip', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<Tooltip>Test</Tooltip>);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <Tooltip color="neutral" content="Test">
                Test
            </Tooltip>,
        );

        expect(tree).toMatchSnapshot();
    });
});
