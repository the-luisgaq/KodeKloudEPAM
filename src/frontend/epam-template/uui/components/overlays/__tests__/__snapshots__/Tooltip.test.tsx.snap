// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should be rendered correctly 1`] = `
<DocumentFragment>
  Test
</DocumentFragment>
`;

exports[`Tooltip should be rendered correctly 2`] = `
<DocumentFragment>
  Test
</DocumentFragment>
`;
